include/mbgl/shaders/ubo_max_count.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <mbgl/shaders/background_layer_ubo.hpp>
+#include <mbgl/shaders/circle_layer_ubo.hpp>
+#include <mbgl/shaders/collision_layer_ubo.hpp>
+#include <mbgl/shaders/debug_layer_ubo.hpp>
+#include <mbgl/shaders/fill_layer_ubo.hpp>
+#include <mbgl/shaders/fill_extrusion_layer_ubo.hpp>
+#include <mbgl/shaders/heatmap_layer_ubo.hpp>
+#include <mbgl/shaders/heatmap_texture_layer_ubo.hpp>
+#include <mbgl/shaders/hillshade_layer_ubo.hpp>
+#include <mbgl/shaders/hillshade_prepare_layer_ubo.hpp>
+#include <mbgl/shaders/line_layer_ubo.hpp>
+#include <mbgl/shaders/raster_layer_ubo.hpp>
+#include <mbgl/shaders/symbol_layer_ubo.hpp>
+
+#include <algorithm>
+
+namespace mbgl {
+namespace shaders {
+
+static constexpr auto maxUBOCountPerShader = std::max({static_cast<size_t>(backgroundUBOCount),
```

Comment:
So we're trading some level of maintainability here for the moment to do this. A follow up for me probably is to generate things like this as part of the revamped shader code generation.

Reply:
Yes, that's true.